vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c

```diff
@@ -833,16 +836,26 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
             }
             else
             {
-                xAdvData->p_manuf_specific_data->data.p_data = IotBle_Malloc( usManufacturerLen );
+                /* extract company_identifier from input manufacturer data
+                because nordic sdk has separate field for company_identifier */
+                uint16_t company_id = 0;
+                for (uint8_t i = 0; i < company_id_size; i++) {
```

Comment:
Coding standard for FreeRTOS is to have the variable declarations at start of the function. This is to maintain portability across different compilers. Could you move all variable definitions accordingly?

Reply:
Moved companyId at the very beginning of the function.